src/styles/utils/_functions.scss

```diff
@@ -0,0 +1,23 @@
+@function calendarSize($cell-size, $gap) {
+  @return ($cell-size * 7) + ($gap * 7);
+}
+
+@for $days from 1 through 31 {
+  .calendar__day:nth-child(#{$days})::before {
+    content: "#{$days}";
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+  }
+}
+
+@for $long-days from 28 through 31 {
+  .calendar__month-length--#{$long-days}>:nth-child(n + #{$long-days}) {
+    display: none;
+  }
+}
+
+@each $name-day, $margin-left in $week {
+  .calendar__start-day--#{$name-day}>:first-child {
+    margin-left: $margin-left;
+  }
+}
```

Comment:
cycles aren't functions and you can place them in the calendar block to use nesting with &